Automattic/wp-calypso
# Jetpack Manage: Onboarding - Adding a two-part add new site guided tour

Related to https://github.com/Automattic/jetpack-manage/issues/159

## Proposed Changes

* This PR adds an 'add new site' onboarding tour in Jetpack Manage. The first part is initiated via the overview page and begins on the 'sites' dashboard - `dashboard`. 
* Pressing on the arrow button next to 'Add New Site' technically 'ends' this tour, as the user would then go about adding a new site ideally. However at the same time it saves a new preference which records the number of sites, used to compare with the number of sites after a new site has been added or connected.
* Once a site has been added, the next part of the tour begins (technically a new tour). It checks the site count in preferences, compares it to the current site count, and if the current site count is bigger then we have our new site (or sites). The tour points to the top of the sites column, as the site could be anywhere on the page (or another page).
* Finally, on clicking 'Got it' a redirect takes you to the 'overview' page.

## Testing Instructions

* You must be an agency to be able to this this PR: 2c49b-pb. Make sure to switch it back to our original type after testing. Or not. Your call!
* Switch to this branch and `yarn start-jetpack-cloud`.
* Open `http://jetpack.cloud.localhost:3000/`, and you'll be redirected to the /dashboard.
* Append the URL with `?tour=add-new-site`
* You should see the first step of the add new sites tour. Clicking the arrow button next to the 'Add New Sites' button should remove the tour (technically finishing it).
* To properly test adding a new site after clicking that button, I'd recommend creating a Jurassic Ninja site, and then following the prompts to connect it to Jetpack.
* **IMPORTANT**: Once connected, you'll need to manually visit ` http://jetpack.cloud.localhost:3000/dashboard?site_connected={site_url}` changing site_url for the site you just connected without the https part (e.g. `http://jetpack.cloud.localhost:3000/dashboard?site_connected=helpless-lizard.jurassic.ninja`).
* You should then see the next stage of the tour, targeting the site you've just connected.
* Clicking on 'Got it' should redirect you to the Overview page.
* Once you finish the tour, to run it again you must erase the preference `jetpack-cloud-site-dashboard-add-new-site-tour`, plus `jetpack-cloud-site-sites-list`. You can do this by hovering in the bottom right button called `Preferences`, then clicking over the `X` next to the preference name.

Preferences:
<img width="1046" alt="Screenshot 2023-12-19 at 14 58 00" src="https://github.com/Automattic/wp-calypso/assets/16754605/0ec75291-868e-4492-8d18-d570f18c25f7">

Tour first step:
<img width="1428" alt="Screenshot 2023-12-22 at 14 23 16" src="https://github.com/Automattic/wp-calypso/assets/16754605/473c3e51-bf4f-418d-9d5c-3c5dd9f5cdac">

Connecting to Jetpack:
<img width="761" alt="image" src="https://github.com/Automattic/wp-calypso/assets/37049295/95369e09-f7d8-405b-8ac2-8bfa2e933f43">

Site connected to Jetpack:
<img width="1396" alt="image" src="https://github.com/Automattic/wp-calypso/assets/37049295/7c3eba4a-3c8c-453a-b36f-eefd0129cb65">

Last step: Tour pointing to the latest site connected to Jetpack (redirect to 'Overview' occurs after clicking the button)
<img width="949" alt="image" src="https://github.com/Automattic/wp-calypso/assets/37049295/2013e101-d0d0-4054-8f13-31f4a05a74a5">


## Pre-merge Checklist
